### PR TITLE
Added vertex cover

### PR DESCRIPTION
**Pull request by Rajat379**
Added a new file which covers the vertex cover of a graph.

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Hey there! Thank you for submitting a PR! 😄 Sadly, this does not meet our requirements. Check out our [repository standards](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md). Please read them **carefully** and **follow them**. **Let us know if you need any help here or on our [Discord server](https://the-algorithms.com/discord).** 🙂

**Comment by Rajat379** [comment]
@Panquesito7 Kindly have a look now, updated the file.

File: graph/vertex_cover.cpp [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    std::list<int> *adj;  // Pointer to an array containing adjacency lists
+ public:
+    Graph(int V);
+    void addEdge(int v, int w);  // function to add an edge to graph
+    void printVertexCover();     // prints vertex cover
+};
+
+}  // namespace
+
+Graph::Graph(int V) {
```
Comment (Panquesito7) [inline graph/vertex_cover.cpp:25]: Functions should be documented explaining the use of its parameters, a brief description, and what it returns.

File: graph/vertex_cover.cpp [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  @author [Rajat Garg](https://github.com/Rajat379)
+  @file
+
+  Implementation of a vertex cover in a graph.
+*/
+
+#include <algorithm>
+#include <cassert>
+#include <iostream>
+#include <list>
```
Comment (Panquesito7) [inline graph/vertex_cover.cpp:11]: Please add a one-line description of what the library/header is for (**see the example below**).

```c++
#include <iostream>    /// for IO operations
#include <cassert>    /// for assert
```

File: graph/vertex_cover.cpp [hunk 3]
```diff
@@ -0,0 +1,83 @@
+/*
+  @author [Rajat Garg](https://github.com/Rajat379)
+  @file
+
+  Implementation of a vertex cover in a graph.
```
Comment (Panquesito7) [inline graph/vertex_cover.cpp:5]: Please describe how the algorithm works, a Wikipedia or any other source for algorithm explanation, what the algorithm does, etc..

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**